app/controllers/my_module_shareable_links_controller.rb

```diff
@@ -58,6 +60,17 @@ def load_my_module
     render_404 unless @my_module
   end
 
+  def load_shareable_link
+    @shareable_link = ShareableLink.find_by(uuid: params[:uuid])
+
+    return render_403 if @shareable_link.blank?
+  end
+
+  def load_asset
+    @asset = Asset.find_signed(params[:id])
```

Comment:
won't it allow downloading of the asset even after the main ShareableLink is deleted?